src/CoreWCF.WebHttp/src/CoreWCF/UriTemplate.cs

```diff
@@ -0,0 +1,1799 @@
+                {
+                    while (relativeSegmentsIndex < numRelativeSegments)
+                    {
+                        string relPathSeg = Uri.UnescapeDataString(candidateSegments[relativeSegmentsIndex + numSegmentsInBaseAddress]);
+                        if (relPathSeg.EndsWith("/", StringComparison.Ordinal))
+                        {
+                            relPathSeg = relPathSeg.Substring(0, relPathSeg.Length - 1); // trim slash
+                        }
+                        relativeSegments.Add(relPathSeg);
+                        relativeSegmentsIndex++;
+                    }
+                }
+                // Checking if we matched all required segments already
+                else if (numMatchedSegments < this._firstOptionalSegment)
+                {
+                    isStillMatch = false;
+                }
+            }
+            else
+            {
+                numMatchedSegments = 0;
+            }
+
+            return isStillMatch;
+        }
+
+        bool IsPathFullyEquivalent(UriTemplate other)
+        {
+            // Refer to the note on template equivalency at IsEquivalentTo
+            // This method checks if both templates has a fully equivalent path.
+            if (HasWildcard != other.HasWildcard)
+            {
+                return false;
+            }
+
+            if (_segments.Count != other._segments.Count)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < _segments.Count; ++i)
+            {
+                if (!_segments[i].IsEquivalentTo(other._segments[i],
+                    (i == _segments.Count - 1) && !this.HasWildcard && (this.IgnoreTrailingSlash || other.IgnoreTrailingSlash)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        BindInformation PrepareBindInformation(IDictionary<string, string> parameters, bool omitDefaults)
+        {
+            if (parameters == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(parameters));
+            }
+
+            IDictionary<string, string> extraParameters = new Dictionary<string, string>(UriTemplateHelpers.GetQueryKeyComparer());
+            foreach (KeyValuePair<string, string> kvp in parameters)
+            {
+                if (string.IsNullOrEmpty(kvp.Key))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgument(nameof(parameters),
+                        SR.Format(SR.UTBindByNameCalledWithEmptyKey));
+                }
+
+                extraParameters.Add(kvp);
+            }
+
+            ProcessDefaultsAndCreateBindInfo(omitDefaults, extraParameters, out BindInformation bindInfo);
+
+            return bindInfo;
+        }
+        BindInformation PrepareBindInformation(NameValueCollection parameters, bool omitDefaults)
+        {
```

Comment:
There's a lot of spacing issues in this file, so I'll make this the last comment for this file to fix them all up.

Reply:
This is addressed in e95ff7c922db5d4a6ee55847b763b49619f08311.